numba/cpython/randomimpl.py

```diff
@@ -1586,3 +1586,60 @@ def dirichlet_impl(alpha, size=None):
         )
 
     return dirichlet_impl
+
+# ------------------------------------------------------------------------
+# np.random.noncentral_chisquare
+
+@overload(np.random.noncentral_chisquare)
+def noncentral_chisquare(df, nonc, size=None):
+
+    @register_jitable
+    def noncentral_chisquare_single(df, nonc):
+        # identical to numpy implementation 
+        
+        if df <= 0:
+            raise ValueError("df <= 0")
+        if nonc < 0:
+            raise ValueError("nonc < 0")
+
+        if np.isnan(nonc):
+            return np.nan
+
+        if df > 1:
```

Comment:
This condition appears reversed in the original NumPy implementation:

https://github.com/numpy/numpy/blob/623bc1fae1d47df24e7f1e29321d0c0ba2771ce0/numpy/random/src/legacy/legacy-distributions.c#L125

Or did I miss something here?

Reply:
See above.